src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -659,7 +659,14 @@ private static bool IsTypeLessVisibleThan(TypeSymbol type, Symbol sym, ref HashS
                         foreach (var typeArg in ((NamedTypeSymbol)current).TypeArgumentsWithAnnotationsNoUseSiteDiagnostics)
                         {
                             // Let's try to avoid early resolution of nullable types
-                            var result = visitType(typeArg);
+                            var result = VisitType(
```

Comment:
>VisitType [](start = 41, length = 9)

It is not clear why we are inlining this call site and leaving two others unchanged. If ```visitType``` captures too much, then the other call sites are affected as well. Why not eliminate the capturing completely?

Reply:
This affects every single recursive call, whereas the function pointer one only affects that function pointer stacks. I can un-inline that one if you prefer, but I didn't consider it necessary.

---
In reply to: [421828630](https://github.com/dotnet/roslyn/pull/44040#discussion_r421828630) [](ancestors = 421828630)

Reply:
I'll address this in a follow up.

---
In reply to: [421835700](https://github.com/dotnet/roslyn/pull/44040#discussion_r421835700) [](ancestors = 421835700,421828630)